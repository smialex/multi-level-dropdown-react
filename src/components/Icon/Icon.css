.icon{
  --icon-size: calc(60px * 0.7);
  width: var(--icon-size);
  height: var(--icon-size);
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: var(--bg-accent);
  border-radius: 50%;
}

.icon > svg {
  width: 20px;
  height: 20px;
  fill: var(--text-color);
}

.icon.icon_transparent {
  background-color: transparent;
}