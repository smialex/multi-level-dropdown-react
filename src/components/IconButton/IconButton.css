.icon-button{
  border: none;
  outline: none;
  background-color: var(--bg);
  border-radius: 50%;
  margin-right: 10px;
  cursor: pointer;
  transition: filter 0.3s ease;
}

.icon-button:hover{
  filter: brightness(1.2);
}