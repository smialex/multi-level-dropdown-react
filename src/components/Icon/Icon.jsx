import React from 'react';
import './Icon.css';

export const Icon = ({icon, transparent}) => {

  const styles = `icon ${transparent ? 'icon_transparent' : ''}`;

  return (
    <div className={styles}>{icon}</div>
  );
}

