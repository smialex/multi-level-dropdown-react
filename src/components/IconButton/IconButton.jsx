import React from 'react';
import './IconButton.css';
import { Icon } from '../Icon';

export const IconButton = ({icon, ...rest}) => {
  return (
    <button className="icon-button" {...rest}>
      <Icon icon={icon}/>
    </button>
  );
}
