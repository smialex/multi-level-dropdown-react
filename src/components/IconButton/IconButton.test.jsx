import React  from 'react';
import renderer from "react-test-renderer";
import { shallow } from 'enzyme';
import { IconButton } from "./IconButton";
import { Icon } from "../Icon";

describe('IconButton component', ()=>{
  test('should render IconButton component', () => {
    const button = renderer.create(<IconButton/>).toJSON();
    expect(button).toMatchSnapshot();
  });

  test('should render Icon component as child', () => {
    const wrapper = shallow(<IconButton/>);
    expect(wrapper.contains(<Icon/>)).toBeTruthy();
  });
  
  test('should handle onClick', () => {
    const onClick = jest.fn()
    const wrapper = shallow(<IconButton onClick={onClick}/>);

    wrapper.find('button').simulate('click');

    expect(onClick).toHaveBeenCalledTimes(1);
  })
  
});