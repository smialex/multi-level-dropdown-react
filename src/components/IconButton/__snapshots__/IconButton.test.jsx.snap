// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton component should be rander IconButton component 1`] = `
<button
  className="icon-button"
>
  <div
    className="icon "
  />
</button>
`;

exports[`IconButton component should rander IconButton component 1`] = `
<button
  className="icon-button"
>
  <div
    className="icon "
  />
</button>
`;

exports[`IconButton component should render IconButton component 1`] = `
<button
  className="icon-button"
>
  <div
    className="icon "
  />
</button>
`;
